sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1105,10 +1105,16 @@ class SessionCatalog(
    * Check if the function with the specified name exists
    */
   def functionExists(name: FunctionIdentifier): Boolean = {
-    val db = formatDatabaseName(name.database.getOrElse(getCurrentDatabase))
-    requireDbExists(db)
-    functionRegistry.functionExists(name) ||
```

Comment:
I think you can do like:

```scala
functionRegistry.functionExists(name) || {
  val db = formatDatabaseName(name.database.getOrElse(getCurrentDatabase))
  requireDbExists(db)
  externalCatalog.functionExists(db, name.funcName)
}
```

Reply:
@HyukjinKwon 's change looks good to me.